docs/technical/README.md

```diff
@@ -1,3 +1,36 @@
-# Technical Design Documentation
+# Technical Documentation
 
-Documents that explain technical design, architecture and interations of various Janssen modules
+Documentation here explains technical design, architecture and interations of various Janssen modules.
+
+Contents:
+
+- [API Reference](#api-reference)
+- [Design Consideration and Guidelines](#design-consideration-and-guidelines)
+  - [REST API Design](#rest-api-design)
+  - [Caching](#caching)
+  - [Testing](#testing)
+  - [Deployment](#deployment)
+  - [Security](#security)
+  - [Scallability and Cloud Infrastructure](#scallability-and-cloud-infrastructure)
+- [Technical Documentation Guidelines](#Technical Documentation Guidelines)
+
+
+## API Reference
+  
+## Design Consideration and Guidelines
+This section outlines high-level design principles, styles and design choices that Janssen project follow. 
+### REST API Design
+### Caching
+### Testing
+### Deployment
+### Security
+### Scallability and Cloud Infrastructure 
+
+## Technical Documentation Guidelines
+  
+Each module directory will hold detailed technical documentation for that module. It should have README file that follows [Technical Overview Template](./technical-overview-template.md). Also create other directories as required and as mentioned in `Directory Structure` section below. 
+
+### Directory Structure
+Create directories within your module's directory as below to support your documetation
+ - `images` : to hold screenshots etc
```

Comment:
Let's specify the image params like size limit . It's very important we don't clog our repo

Reply:
About clogging of repo, should we rethink about what I have proposed in this PR with respect to _where we store our media assets?_ 

In this PR, I have put media assets under `docs` directory. Similar media assets would exist in `user` documentation and `developer` documentation. Yes, we can put recommendation, but there is no way to stop people from uploading larger files (GH allows up to 100MB). Plus, over time, we will add GIF(tutorials), or may be video files for that matter. If we keep media assets at different places, it'll hard to control, review and clean-up. 

So I am having second thoughts about this arrangement. Now I think that keeping all media assets in a separate top-level directory (`jans/media-assets`) would be better. It'll be easy to keep in check the size of one directory. Also, maybe later we can add automation using GH actions where we can have an action trigger on commit in `jans/media-assets`. That action can make sure the newly added asset is in-line with recommended size and type etc.

Reply:
I think we should keep all assets in the docs and we shouldn't break it off. As for the size and type just mention that in the docs recs so people know. Its the responsibility of the reviewer of the docs PR to make sure the image/diagram is of spec. We will add a todo to check the assets directories in a PR to make sure it meets the specs we specified.

Two options:
1. Each module will have a predefined set of folders such as `images` and `diagrams`.
2. You have one folder  `docs/assets` that contains all the modules and their folders `images` and `diagram` .

I would go with the first option and hence you need to add empty folders for each module. 

The action we will add will check recursively for any folder that has the name `images` or `diagram`

Reply:
I see few problems if we go for multiple directories:
- directories outside of the `docs` may also require to store media files.
- In the future, new directories will have to be created. Like `videos`, `GIFs` etc. And all this will happen in various directories. 
- Yes, PR reviewers should ideally be vigilant about the size fo the uploaded files, but it is highly likely to be missed. Automated checks will only prove to be working in long run
- Cleaning up would be difficult as it'll be difficult to understand what is the total size of media being held in repo as media will be scattered. 
- And it'll be an additional risk that GH actions would fail if someone has named the directory `image` instead of `images`. Controlling naming standards would be another problem. Instead, if everything is in one place, there will be only one `images` directory and that is it. Developers will just have drop their images in there. Not hassle of creating new directories. Easy.

Reply:
I am good with option 2. It is a middle path. 
In your description of option 2 above, are you suggesting that we should have a folder for each module under `docs/assets`? I think that is not required because there is no benefit in separating images by modules. `docs/assests` can just have two folders `images` and `diagrams` and that is it. Images for all modules stay together. And developers can refer to them in their documentation. Do you see benefit of separating them by modules?

Reply:
yes I think they should be separated by module. The likelihood that they asset is related to a module ( a service in the module) is high. That way we can offer a bit more distinction where these assets are being used . So take a look at it in from a reverse standpoint a document points to a link but with separate modules we also tie the assets to the module. Looking where to edit or where to fix will be a lot easier especially if we have a lot. We also provide a safe guard against miss use that way and drive adoption. Easier for the dev to find so easier to write a doc.  I also want to do this in case later on some of these assets get inherited or used by code. Doing this will certainly help. Otherwise we can prefix with `service_` such as `jans_auth_server_*`. We also need a clear nomenclature for the images. This helps with automation , cleanup ...etc so it should be added to this doc.

Reply:
Agreed. :+1: 
I'll make changes in the instructions accordingly.

Reply:
Refactored the instructions. Please review.